lib/base-compiler.ts

```diff
@@ -378,12 +378,16 @@ export class BaseCompiler implements ICompiler {
             options.ldPath = this.getSharedLibraryPathsAsLdLibraryPaths([]);
         }
 
+        // Take a (shallow) copy of the options before we add a random customCwd: The fact we have createAndUseTempDir
+        // set is enough to make us different from an otherwise identical run without createAndUseTempDir. However, the
+        // actual random path is unimportant for caching; and its presence prevents cache hits.
+        const optionsForCache = {...options};
         if (options.createAndUseTempDir) {
-            const tmpDir = await this.newTempDir();
-            options.customCwd = tmpDir;
+            options.customCwd = await this.newTempDir();
         }
 
-        const key = this.getCompilerCacheKey(compiler, args, options);
+        const key = this.getCompilerCacheKey(compiler, args, optionsForCache);
+        if (options.createAndUseTempDir) delete key.options.customCwd;
```

Comment:
Isn’t this redundant? Its already using the key before options was extended with customcwd

Reply:
Oh shucks, good spot. Yeah it absolutely is and in fact is _the point_ of me taking a copy before....thanks!